client/src/main/scala/drt/client/services/SPACircuit.scala

```diff
@@ -269,7 +269,7 @@ class CrunchUpdatesHandler[M](airportConfigPot: () => Pot[AirportConfig],
 
   def updateAndTrimFlights(crunchUpdates: CrunchUpdates, existingState: CrunchState, keepFromMillis: MillisSinceEpoch): Set[ApiFlightWithSplits] = {
     val thirtyMinutesMillis = 30 * 60000
-    val relevantFlights = existingState.flights.filter(keepFromMillis - thirtyMinutesMillis <= _.apiFlight.PcpTime)
+    val relevantFlights = existingState.flights.filter(keepFromMillis - thirtyMinutesMillis <= _.apiFlight.PcpTime.getOrElse(0L))
```

Comment:
Could we rather filter out PcpTime of None either as part of the condition, or a separate collect, or filter.